code/datums/beam.dm

```diff
@@ -134,16 +145,136 @@
 		X.pixel_y = Pixel_y
 		CHECK_TICK
 
+/datum/beam/laser
+	///Is TRUE when the beam is not from a player (like a sentry); if TRUE, will track targets
```

Comment:
From what i can tell this subtype shouldnt exist and this can be handled as args or flags on the parent or should be handled by the gun firing anyway

Reply:
Beam code looks like an incomprehensible mess to me and didn't want to break normal beams by making this. I don't see why the beam re-drawing should be handled by the source object when the beam should be doing everything.

Reply:
Well copy pasting beam code is a downgrade so I'm afraid you're gonna have to make an effort to integrate it

Reply:
Also idk how the source object relates this is the datum